tests/functional/aws-node-sdk/test/object/completeMPU.js

```diff
@@ -78,9 +50,115 @@ describe('Complete MPU', () => {
                     checkNoError(err);
                     // to show that data in completed key is just first part
                     assert.strictEqual(data.ContentLength, '10');
-                    done();
+                    if (versionId) {
+                        assert.strictEqual(data.VersionId, versionId);
+                    }
+                    cb();
                 });
             });
+        }
+
+        afterEach(done => {
+            removeAllVersions({ Bucket: bucket }, err => {
+                if (err) {
+                    return done(err);
+                }
+                return s3.deleteBucket({ Bucket: bucket }, done);
+            });
+        });
+
+        describe('on bucket without versioning configuration', () => {
+            let uploadId;
+            let firstEtag;
+
+            beforeEach(() => s3.createBucketAsync({ Bucket: bucket })
```

Comment:
The setup in beforeEach is more or less the same across the tests. Can you extract this into a utility and use it across the tests?